packages/pulumi-aws/src/apps/storage/StorageApp.ts

```diff
@@ -0,0 +1,66 @@
+import {
+    defineApp,
+    createGenericApplication,
+    ApplicationContext,
+    ApplicationHooks
+} from "@webiny/pulumi-sdk";
+
+import { createCognitoResources } from "./StorageCognito";
+import { createDynamoTable } from "./StorageDynamo";
+import { createFileManagerBucket } from "./StorageFileManager";
+
+export interface StorageAppConfig extends Partial<ApplicationHooks> {
+    config?(app: StorageApp, ctx: ApplicationContext): void;
+    protect?(ctx: ApplicationContext): boolean;
+}
+
+export const StorageApp = defineApp({
+    name: "Storage",
+    config(app, config: StorageAppConfig) {
+        const protect = config.protect?.(app.ctx) ?? app.ctx.env === "dev";
```

Comment:
`dev` should be `prod` here I think?

Reply:
Thanks for finding it :) 
It should of course be `!==`. 

That is the question, do we assume, that by default there is one `prod` environment that is protected and all other are considered non-production, or we assume that there is `dev` environment, and all other are considered production?

Reply:
The first seems to be more common, but the latter is safer.

Reply:
As it is today, we assume all envs to be `dev` unless it's a `prod` env, but the developers can control that in their current project setups. With the new setup, I was looking at resolving it with that map I showed you, remember this?
```
devEnvironments: ["dev", "test"],
prodEnvironment: ["prod", "staging", "client"]
```

I'm glad we landed on this topic, and we can now discuss how do we want to approach this. Feel free to lay out your ideas on this topic.